tool/test/start-core-server.sh

```diff
@@ -25,4 +25,5 @@ set -e
 rm -rf typedb-all
 
 bazel run //tool/test:typedb-extractor -- typedb-all
+JAVA_HOME=$(bazel run //tool/test:echo-java-home)
```

Comment:
set `JAVA_HOME` for typedb process launched below.

This feels a little flimsy, since it points inside a bazel folder. e.g: 
`/private/var/tmp/_bazel_krishnangovindraj/5096250ab346cdc7499cfe043de9e352/external/remotejdk11_macos_aarch64/zulu-11.jdk/Contents/Home`